src/rpc/blockchain.cpp

```diff
@@ -1515,8 +1514,18 @@ UniValue getchaintxstats(const JSONRPCRequest& request)
         }
     }
 
-    if (blockcount < 1 || blockcount >= pindex->nHeight) {
-        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid block count: should be between 1 and the block's height");
+    if (pindex->nHeight < 2) {
+        blockcount = 0;
+    } else {
+        blockcount = std::min(blockcount, pindex->nHeight - 1);
+    }
+
+    if (!request.params[0].isNull()) {
+        blockcount = request.params[0].get_int();
+
+        if (blockcount < 1 || blockcount >= pindex->nHeight) {
```

Comment:
Any reason to disallow 0? `blockcount` can be 0 if we run this on a node with just the genesis block, so why not allow it to be set to 0 when running with a longer chain?